consensus/ethash/ethash.go

```diff
@@ -320,15 +320,31 @@ func MakeDataset(block uint64, dir string) {
 	d.release()
 }
 
+type Mode uint
+
+const (
+	Normal Mode = iota
+	Shared
+	Test
+	Fake
+	FullFake
+)
```

Comment:
Please prefix these with `Mode` so that they are:

 * Semantically grouped together similar to e.g. https://golang.org/pkg/net/http/#pkg-constants
 * Easier to understand `ethash.ModeTest` vs. `ethash.Test`